internal/app/synchronizer/evaluator_client.go

```diff
@@ -99,23 +99,29 @@ func newGrpcEvaluator(cfg config.View) (evaluator, error) {
 	}, nil
 }
 
-func (ec *grcpEvaluatorClient) evaluate(ctx context.Context, proposals []*pb.Match) ([]*pb.Match, error) {
+func (ec *grcpEvaluatorClient) evaluate(ctx context.Context, pc <-chan []*pb.Match) ([]*pb.Match, error) {
```

Comment:
// at start
sendError := make(chan error, 1)

// before return in the go routine:
sendError <- err

// before end
err = <-sendError
if err != nil {
  return nil, err
}
return results, nil

Reply:
oh also send nil to sendError at end of go routine

Reply:
I made a slight twist to it to have this change align with the pattern that i used to handle the errors in http calls at line 168, could review this PR again?